### ISSUE TITLE
Fails to start with "Waiting for release of fcntl lock"

### ISSUE THREAD

**Issue by oddhack** [SLASHEM-Extended/SLASHEM-Extended#18]
Building on Debian 9.1 from commit f3d16ae8024c1e46fba74020283ead03913fdf20, following instructions on https://nethackwiki.com/wiki/Slash'EM_Extended :

> sh sys/unix/setup.sh
> make -f sys/unix/GNUmakefile
> make -f sys/unix/GNUmakefile install

This installs to $HOME/slex/slex-2.0.3 . Then executing

$HOME/slex/slex-2.0.3/slex

results in

> This is SuperLotsoAddedStuffHack-Extended Magic Extended 1997-2016
> NetHack, Copyright 1985-2003 Stichting Mathematisch Centrum, M. Stephenson.
> Slash'EM Extended version by Amy (user Bluescreenofdeath at Nethackwiki).
> See license for details. Bug reports to Bluescreenofdeath's user talk page.
> Cannot open file perm. This is a program bug.
> Waiting for release of fcntl lock on perm. (9 retries left).
> Waiting for release of fcntl lock on perm. (8 retries left).
> Waiting for release of fcntl lock on perm. (7 retries left).
> Waiting for release of fcntl lock on perm. (6 retries left).
> Waiting for release of fcntl lock on perm. (5 retries left).
> Waiting for release of fcntl lock on perm. (4 retries left).
> Waiting for release of fcntl lock on perm. (3 retries left).
> Waiting for release of fcntl lock on perm. (2 retries left).
> Waiting for release of fcntl lock on perm. (1 retries left).
> Waiting for release of fcntl lock on perm. (0 retries left).
> I give up.  Sorry.
> Some other process has an unnatural grip on perm.
>
> Hit space to continue:

Removing the file 'perm' in $HOME/slex/slex-2.0.3/perm and trying again doesn't help. I'm wondering if it's trying to lock a file outside the actual installation directory? Or got built with the wrong file locking path?

**Comment by oddhack** [comment]
Also, I'm unclear on whether bug reports should be filed here or on nethackwiki? There seems to be a lot of commit activity here but very little issue activity so maybe I'm in the wrong place to report a problem?

**Comment by moon-chilled** [comment]
Here is fine.  Is there a file named perm_lock or something_lock in $home/slex/slex-2.0.3?  Also, if you're using GNUmakefile, then you don't have to do `sh sys/unix/setup.sh`

**Comment by oddhack** [comment]
There is no perm_lock . Unfortunately the error message isn't clear on what file it's *trying* to lock, and I am slightly suspicious that it's trying to lock something not in the installed directory hierarchy. I saw the same thing when building SlashTHEM yesterday and found that by reverting to the default top-level Makefile UID/GID/install path, the error went away - haven't tried that here, yet.

Building seems poorly documented. Most of the top-level documentation in git appears to date to Nethack or Slash'EM days and hasn't been updated - there's a pointer to sys/unix/Install.unx but that dates to NH 3.4. I referred to the Wiki instructions at https://nethackwiki.com/wiki/Slash'EM_Extended#Compiling_on_unix as the project owner here seems to be pretty active there - but if there's more up-to-date build documentation I'd love a pointer to it.

**Comment by moon-chilled** [comment]
Yeah, the unix makefiles generally work a lot better than the GNUmakefiles (sorry @kerio92).  Try doing that with slex.

**Comment by oddhack** [comment]
@Elronnd I did try using the top-level Makefile, changing PREFIX=/usr , GAMEUID = myuid, GAMEGRP = mygrp. "make" succeeds, but "sudo make install" fails with

```
( cd dat ; cp  dungeon dungeon2 dungeon3 dungeon4 dungeon5 dungeon6 dungeon7 dungeon8 asmod-?.lev baalz-?.lev baalz-??.lev bigrm-?.lev castl-?.lev fakewiz?.lev bigrm-??.lev compu.lev exmall-?.lev forge.lev heck-?.lev heck-??.lev hllma-?.lev hitch.lev intpla-?.lev key.lev palace_e.lev palace_f.lev rndmaze.lev rndmazea.lev school-?.lev schoolX.lev sheolfil.lev sheolmid.lev soko?-??.lev town-?.lev unhck-?.lev unhck-??.lev juibl-?.lev knox-?.lev medus-?.lev mnend-?.lev minefil?.lev castl-??.lev mnend-??.lev hmnen-??.lev levgn-?.lev levgn-??.lev levgx-?.lev levgx-??.lev hellfil?.lev hmint-?.lev hmint-??.lev hmnen-?.lev hking-?.lev hking-??.lev oracleX.lev erats-?.lev hrats-?.lev hstor-?.lev mking-??.lev eking-??.lev mintn-??.lev intplx-?.lev rndmazeb.lev mintn-?.lev oracle.lev orcus-?.lev sanct-?.lev soko?-?.lev tower?.lev valley-?.lev wizard?.lev ghbgr-?.lev ghbgr-??.lev astral.lev air.lev earth.lev fire.lev water.lev eking-?.lev esheo-?.lev hsheo-?.lev ???-?.lev mazes-?.lev mazes-??.lev emtemple.lev hmtemple.lev ecav2.lev hcav2.lev efrnk.lev hfrnk.lev etomb.lev htomb.lev espiders.lev hspiders.lev esea.lev hsea.lev egrund-?.lev hgrund-?.lev eknox-?.lev hknox-?.lev hdragons.lev forgeX.lev hitchX.lev compuX.lev keyX.lev etown-?.lev htown-?.lev egehn-?.lev egehn-??.lev hgehn-?.lev hgehn-??.lev dispa-??.lev asmod-??.lev hitchE.lev compuE.lev keyE.lev emedu-?.lev emedu-??.lev hmedu-?.lev hmedu-??.lev ecstl-?.lev ecstl-??.lev hcstl-?.lev ekobol-?.lev hkobol-?.lev enymp-?.lev hnymp-?.lev hcstl-??.lev behol-?.lev blkmar.lev cav2fill.lev medus-??.lev emptb-?.lev emptb-??.lev grund-?.lev demog-?.lev dispa-?.lev dragons.lev geryo-?.lev guild-?.lev kobold-?.lev lich-?.lev mall-?.lev mall.lev mking-?.lev mtemple.lev nymph-?.lev rats-?.lev sea.lev spiders.lev stor-?.lev juibl-??.lev yeeno-??.lev orcus-??.lev demog-??.lev geryo-??.lev tomb.lev yeeno-?.lev frnknstn.lev night-?.lev hllma-??.lev  ???-goal.lev ???-fil?.lev ???-loca.lev ???-strt.lev /usr/local/slexdir )
cp: cannot stat 'asmod-??.lev': No such file or directory
cp: cannot stat 'juibl-??.lev': No such file or directory
cp: cannot stat 'orcus-??.lev': No such file or directory
cp: cannot stat 'demog-??.lev': No such file or directory
cp: cannot stat 'geryo-??.lev': No such file or directory
Makefile:376: recipe for target 'dofiles-nodlb' failed
make[2]: *** [dofiles-nodlb] Error 1
```

Maybe it would be better to use prefix-*.lev as the double-digit levels for these special levels don't exist.

**Comment by oddhack** [comment]
OK, after editing out the offending 'demoname-??.lev' bits from Makefile, I did successfully install using the top-level Makefile. Unfortunately, exactly the same behavior results - can't grab a lock on perm, tries for a while and gives up.

Is there an alternate file locking method I could enable in the build? I'm not going to be running it multiuser / multisession in any event, so not being able to grab a lock at least won't create a race condition.

**Comment by RikerW** [comment]
Workaround for right now: try navigating to dnethackdir (inside the slex main directory), and then running ./dnethack.  That should work.

**Comment by AmyBSOD** [comment]
"dnethackdir"??? "./dnethack"??? That sounds to me like it should not work, considering this isn't dnethack :D

**Comment by RikerW** [comment]
_sigh_, bad copy-paste.  Just tested, works with slex too though.  I was in the wrong directory at the time.

Rewritten: try navigating to the created folder after you run `make install`.   Inside, it should have a `slex` binary or executable, run `./slex` while in that directory.  It's a hacky workaround, but it works.
